Makefile

```diff
@@ -46,7 +46,7 @@ $(GOLANGCI):
 
 lint: .lint
 
-.lint: $(ALL_GO_FILES) $(GOLANGCI) .golangci.yml
+.lint: $(ALL_GO_FILES) $(GOLANGCI)
```

Comment:
why drop the config altogether? should we do that in atomix also?

Reply:
There were like 20 lines and I didn't really want to figure out which of those were deprecated or if we liked them. The defaults seemed sane, and we already have a few `//nolint:` lines that we're not fixing.

I'm open to either an updated golangci.yml but prefer someones curated one rather than figuring out that myself.

Reply:
ok, makes sense. maybe I will try the default on atomix and see what it comes up with. I had the same feeling of being unsure which ones to leave in etc, when looking at this for atomix